main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -84,8 +84,7 @@ object CompletionProvider {
       //
       // Types.
       //
-      case SyntacticContext.Type.Eff => EffSymCompleter.getCompletions()
-      case SyntacticContext.Type.OtherType => TypeCompleter.getCompletions(ctx) ++ EffSymCompleter.getCompletions()
```

Comment:
Also, are we sure we are covering all cases here?

Reply:
Yes, the only way a `SyntacticContext.Type.*` is built is from `UndefinedType`.
https://github.com/flix/flix/blob/983766f6689ad454e6d4f1d5757d154d15f62dc9/main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala#L216C1-L216C93
So we just move the EffSymCompleter to SemanticCompleter. As a result, EffSymCompleter can access the `err.qn` and tell if the undefined type is qualified.